connection.go

```diff
@@ -54,12 +56,12 @@ type Route struct {
 
 // Connection is a wrapper around the actual amqp.Connection and amqp.Channel
 type Connection struct {
-	started     bool
-	serviceName string
-	amqpUri     amqp.URI
-	connection  amqpConnection
-	channel     AmqpChannel
-	handlers    map[queueRoutingKey]messageHandlerInvoker
+	started       bool
+	serviceName   string
+	amqpUri       amqp.URI
+	connection    amqpConnection
+	channel       AmqpChannel
+	queueHandlers handlers.QueueHandlers[messageHandlerInvoker]
```

Comment:
Will we ever have any other type to QueueHandlers or could we move that into the type itself?

Reply:
`QueueHandlers` is defined in another package...we would have to move `messageHandlerInvoker` then as well..and `messageHandlerInvoker` contains `HandlerFunc` so we will get a circular dependency unless we move it to a separate package as well. Not a big problem of course.
And the short answer to the question is "No, we will never have any other type" 😊

Reply:
True. Not worth the effort right now then.